src/overlays/gamestates/ovl_file_choose/z_file_choose.c

```diff
@@ -1488,16 +1488,15 @@ void FileChoose_LoadGame(GameState* thisx) {
         gSaveContext.buttonStatus[3] = gSaveContext.buttonStatus[4] = BTN_ENABLED;
 
     gSaveContext.unk_13E7 = gSaveContext.unk_13E8 = gSaveContext.unk_13EA = gSaveContext.unk_13EC =
-        gSaveContext.unk_13F4 = 0;
+        gSaveContext.magicCapacity = 0;
 
-    gSaveContext.unk_13F6 = gSaveContext.magic;
-    gSaveContext.magic = 0;
-    gSaveContext.magicLevel = gSaveContext.magic;
-
-    if (1) {}
+    // Set the fill target to be the magic from the previous save
+    gSaveContext.magicFillTarget = gSaveContext.magic;
+    // Set magicLevel and magic to 0 so the magic meter and magic grows from nothing to the state before the save
```

Comment:
```suggestion
    // Set `magicLevel` and `magic` to 0 so `magicCapacity` then `magic` grows from nothing to respectively the full capacity and `magicFillTarget`
```
it's a lower level description idk 🤔